python/lsst/obs/base/cli/butler_cmd_test.py

```diff
@@ -36,24 +36,41 @@ class ButlerCmdTestBase(metaclass=abc.ABCMeta):
     Subclass from this, then `unittest.TestCase` to get a working test suite.
     """
 
-    instrument_class = None
-    """The fully qualified instrument class.
-    """
+    @staticmethod
+    @abc.abstractmethod
+    def instrumentClass():
+        """Get the fully qualified instrument class.
+
+        Returns
+        -------
+        `str`
+            The fully qualified instrument class.
+        """
+        pass
+
+    @staticmethod
+    @abc.abstractmethod
+    def instrumentName():
```

Comment:
Given that the Instrument class itself defines the name, wouldn't it be easier for everyone if a subclass could specify an Instrument class directly (not a string, although a string could be doImported) and the name was retrieve from that directly? It would prevent an obvious case of duplication of information in the tests where the test subclass now needs to specify the name and the class name and ensure that they all match with the name that they already know is defined in the Instrument class.